rust/connlib/tunnel/src/tests.rs

```diff
@@ -590,27 +599,68 @@ impl ReferenceStateMachine for ReferenceState {
                         state.client_connected_cidr_resources.insert(resource);
                     }
                     Some(_) | None => {
-                        // Depending on the DNS query type, we filter the resolved addresses.
-                        let ips_resolved_by_query = all_ips.iter().copied().filter({
-                            #[allow(clippy::wildcard_enum_match_arm)]
-                            match r_type {
-                                RecordType::A => {
-                                    &(|ip: &IpAddr| ip.is_ipv4()) as &dyn Fn(&IpAddr) -> bool
-                                }
-                                RecordType::AAAA => {
-                                    &(|ip: &IpAddr| ip.is_ipv6()) as &dyn Fn(&IpAddr) -> bool
+                        // If we are NOT connected to this resource yet, we will connect and expire existing DNS records in the process.
+                        if !state.client_dns_records.contains_key(&domain) {
+                            for ip in &all_ips {
+                                for (prev_domain, prev_ips) in state.client_dns_records.iter_mut() {
+                                    prev_ips.retain(|(prev_ip, _)| {
+                                        if prev_ip == ip {
+                                            tracing::debug!(
+                                                "Expiring DNS record: '{prev_domain} -> {prev_ip}'"
```

Comment:
I'm surprised 10 indents doesn't set off clippy lol

Reply:
```suggestion
                                                "Evicting DNS record: '{prev_domain} -> {prev_ip}'"
```
"Expire" makes me think it's based on timing but we're removing the record because a newer record needs that same IP, right?

If the user repeatedly requests `foo.example.com` and `bar.example.com` which have the same IP, it will just keep evicting the entries and replacing them?